src/main/java/com/example/demo/contrtoller/BookController.java

```diff
@@ -13,18 +15,19 @@
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 @RequiredArgsConstructor
 @RestController
-@RequestMapping("/books")
+@RequestMapping("/api/books")
```

Comment:
add `server.servlet.context-path=/api` to application.properties

```suggestion
@RequestMapping("/books")
```